### PR TITLE
network: handle IPv6 routes with reject type correctly

### PR DESCRIPTION
**Pull request by yuwata**
Inspired by #17831.

### CODE REVIEW

File: test/test-network/systemd-networkd-tests.py [hunk 1]
```diff
@@ -2242,6 +2242,7 @@ def test_route_static(self):
 
         copy_unit_to_networkd_unit_path('25-address-static.network')
         check_output(*networkctl_cmd, 'reload', env=env)
+        time.sleep(1)
```
Comment (keszybz) [inline test/test-network/systemd-networkd-tests.py:2260]: Hmm, is `reload` asynchronous? I thought it does a (synchronous) bus call?

**Review by keszybz — COMMENTED** [review]
Looks reasonable, but I don't grok the comment on the first patch.

File: test/test-network/systemd-networkd-tests.py (hunk 1 above)
Comment (yuwata) [reply test/test-network/systemd-networkd-tests.py:2260]: Sorry, the explanation was incorrect. The operation reloading .network files are synchronous. But subsequent reconfiguration of network interfaces are asynchronous.

**Comment by yuwata** [comment]
@keszybz Thank you for the review. Force pushed a new version. Nothing is changed, except for the incorrect commit message. PTAL.

**Comment by keszybz** [comment]
centos failed:
```
08:07:11 ======================================================================
08:07:11 FAIL: test_route_static (__main__.NetworkdNetworkTests)
08:07:11 ----------------------------------------------------------------------
08:07:11 Traceback (most recent call last):
08:07:11   File "./test/test-network/systemd-networkd-tests.py", line 2222, in test_route_static
08:07:11     self.assertIn('blackhole 2001:1234:5678::2 dev lo proto static metric 1024 pref medium', output)
08:07:11 AssertionError: 'blackhole 2001:1234:5678::2 dev lo proto static metric 1024 pref medium' not found in 'blackhole 2001:1234:5678::2 dev lo proto static metric 1024 error -22 pref medium'
```

**Comment by yuwata** [comment]
```
### ip -6 route show type blackhole
blackhole 2001:1234:5678::2 dev lo proto static metric 1024 error -22 pref medium
```
Hmm...

**Comment by yuwata** [comment]
Updated. Now all `metric 1024 pref medium` checks are dropped.

**Comment by keszybz** [comment]
centos passed now. Let's merge.

**Pull request merged**